test/functional/features/steps/openshift.py

```diff
@@ -0,0 +1,278 @@
+                        f'oc get {resource_type} {name} -n {namespace} -o "jsonpath={json_path}"')
+                    attempts -= 1
+                    time.sleep(5)
+        exit_code | should.be_equal_to(0).desc(f'Exit code should be 0:\n OUTPUT:\n{output}')
+        return output
+
+    def get_resource_info_by_jq(self, resource_type, name, namespace, jq_expression, wait=False):
+        output, exit_code = self.cmd.run(
+            f'oc get {resource_type} {name} -n {namespace} -o json | jq -rc \'{jq_expression}\'')
+        if exit_code != 0:
+            if wait:
+                attempts = 5
+                while exit_code != 0 and attempts > 0:
+                    output, exit_code = self.cmd.run(
+                        f'oc get {resource_type} {name} -n {namespace} -o json | jq -rc \'{jq_expression}\'')
+                    attempts -= 1
+                    time.sleep(5)
+        exit_code | should.be_equal_to(0).desc(f'Exit code should be 0:\n OUTPUT:\n{output}')
+        return output.rstrip("\n")
+    
+    def exec_container_in_pod(self, container_name, pod_name, container_cmd):
+        cmd = f'oc exec {pod_name} -c {container_name} {container_cmd}'
+        output, exit_status = self.cmd.run(cmd)
+        print(f"Inside the {pod_name} container {container_name}: {output}, {exit_status}")
+        if exit_status == 0:
+            return output
+        return None
+    
+    def exec_in_pod(self, pod_name, container_cmd):
+        cmd = f'oc exec {pod_name} -- {container_cmd}'
+        output, exit_status = self.cmd.run(cmd)
+        print(f"Inside the {pod_name}: {output}, {exit_status}")
+        if exit_status == 0:
+            return output
+        return None
+    
+    def oc_process_template(self,file_path):
+        cmd = f'oc process -f {file_path}|oc create -f -'
+        output, exit_status = self.cmd.run(cmd)
+        print(f"Proccesing {file_path} template with : {output}")
+        if exit_status == 0:
+            return output
+        return None
+
+    def delete(self,resource_type: str,resource: str, namespace: str):
+        '''
+        Delete resources in a specific namespace
+        '''
+        cmd = f'oc delete {resource_type} {resource} -n {namespace}'
+        output, exit_status = self.cmd.run(cmd)
+        print(f"{output}")
+        if exit_status == 0:
+            return output
+        return None
+
+    def getmasterpod(self,namespace: str)-> str:
+        '''
+        returns the jenkins master pod name
+        '''
+        deploy_pod = 'jenkins-1-deploy'
+        master_pod = ''
+        pods = self.get_pod_lst(namespace)
+        podList = list(pods.split(" "))
+        for pod in podList:
+            if pod == deploy_pod:
+                podList.remove(pod)
+            elif 'jenkins-1-' in pod and deploy_pod not in pod:
+                master_pod = pod
+        return master_pod
+    
+    def scaleReplicas(self, namespace: str, replicas: int, rep_controller: str):
```

Comment:
should be renamed to `scale_replicas` or `scale_replication_set` or just `scale`